### PR TITLE
Feature/kyv 105 filtering available payment methdods

### PR DESCRIPTION
**Pull request by karoliina-elo**

### CODE REVIEW

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/OnlinePaymentController.java [hunk 1]
```diff
@@ -2,10 +2,7 @@
 
 import fi.hel.verkkokauppa.common.error.CommonApiException;
 import fi.hel.verkkokauppa.common.error.Error;
-import fi.hel.verkkokauppa.payment.api.data.GetPaymentMethodListRequest;
-import fi.hel.verkkokauppa.payment.api.data.GetPaymentRequestDataDto;
-import fi.hel.verkkokauppa.payment.api.data.PaymentMethodDto;
-import fi.hel.verkkokauppa.payment.api.data.PaymentReturnDto;
+import fi.hel.verkkokauppa.payment.api.data.*;
```
Comment (hleisma) [inline paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/OnlinePaymentController.java:5]: In general should avoid star imports (could be caused by IDE here). You wont know if there are 1 or 100 imports and what other dependencies the other classes cause.

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/data/PaymentMethodFilter.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum PaymentMethodFilter {
+
+    // If namespace has multiple filters, all available methods for all will be returned
+    ORDER_TYPE(Collections.singletonList("asukaspysakointi")),
```
Comment (hleisma) [inline paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/data/PaymentMethodFilter.java:11]: If we have fixed filters (which means they can be referred to from properties easily with a single property), we should use env properties to select for whom they are active.

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodListService.java [hunk 3]
```diff
@@ -33,4 +40,62 @@
...
+				new Error("unknown-payment-method-filter-type", "unknown payment method filter type")
+		);
+	}
+
+	// namespace can be used as a parameter if different namespaces need to return different values for filter type
+	public HashMap<String, List<String>> getFilterValuesMap(PaymentMethodFilter filter) {
+		HashMap<String, List<String>> values = new HashMap<>();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			values.put("subscription", Collections.singletonList("creditcards"));
```
Comment (hleisma) [inline paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodListService.java:95]: The available keywords that can be used as filters (and are specific to Visma response) should be listed as constants. We should probably have a keywords/constants section in common package for order type.

**Review by hleisma — APPROVED** [review]

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodListService.java (hunk 3 above)
Comment (severikupari1) [reply paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodListService.java:95]: You can make one for PaymentType.CREDIT_CARDS = "creditcards"

File: paymentapi/src/main/resources/application.properties [hunk 4]
```diff
@@ -20,3 +20,6 @@ payment_default_return_url=http://localhost:8080/
 payment_default_notify_url=http://localhost:8080/
 payment_default_submerchant_id=36240
 payment_default_cp=PRO-31312-1
+
+# namespaces separated by comma
+enabled_namespaces.payment_method_filter.ORDER_TYPE=asukaspysakointi
```
Comment (severikupari1) [inline paymentapi/src/main/resources/application.properties:25]: Should this config come from configuration settings and not from environment variable? These could be then easier managed, no need to new release every time values needs to be changes?

**Pull request merged**